handshake_message_certificate.go

```diff
@@ -1,28 +1,34 @@
 package dtls
 
-import (
-	"crypto/x509"
-)
-
 type handshakeMessageCertificate struct {
-	certificate *x509.Certificate
+	certificate [][]byte
 }
 
 func (h handshakeMessageCertificate) handshakeType() handshakeType {
 	return handshakeTypeCertificate
 }
 
 func (h *handshakeMessageCertificate) Marshal() ([]byte, error) {
-	var raw []byte
-	if h.certificate != nil {
-		raw = h.certificate.Raw
+	certTotalLen := 0
+	for _, r := range h.certificate {
+		certTotalLen += len(r)
 	}
-
-	out := make([]byte, 6)
-	putBigEndianUint24(out, uint32(len(raw))+3)
-	putBigEndianUint24(out[3:], uint32(len(raw)))
-
-	return append(out, raw...), nil
+	outLen := len(h.certificate)*3 + certTotalLen
```

Comment:
A few magic numbers in this function (i.e. 3 and 6 mainly).  Should this be replaced with const integers that illustrate what the actual values mean?

Reply:
Fixed!